### PR TITLE
fix(logger): test generates logfile

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

- Currently test_copy_config_to_ext_loggers_clean_old_handlers creates a file called "logfile"

## Description of changes:

- Use NullHandler instead of FileHandler
- Update test codes
- Use `isinstance` over `type`

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by michaelbrewer** [comment]
https://github.com/awslabs/aws-lambda-powertools-python/pull/937#pullrequestreview-849461710

@heitorlessa - So this is why :)

**Comment by michaelbrewer** [comment]
@mploski  - I assume for the tests I can replace `logging.FileHandler` with `logging.NullHandler`, because the former is generating a file after the tests complete

**Review by heitorlessa — APPROVED** [review]

**Comment by heitorlessa** [comment]
@mploski could you clarify what this test tests?

https://github.com/awslabs/aws-lambda-powertools-python/pull/971/files#diff-be72cf49c5b9ee9da9930ced7b33c6112dcc741221f3dfa0579b99e15fc8e8ffR145

I'm not entirely sure, perhaps what the `THEN` line should look like?

**Comment by mploski** [comment]
> @mploski - I assume for the tests I can replace `logging.FileHandler` with `logging.NullHandler`, because the former is generating a file after the tests complete

thats correct @michaelbrewer  we can use nullhandler or stream handler for example

**Comment by mploski** [comment]
> @mploski could you clarify what this test tests?
> 
> https://github.com/awslabs/aws-lambda-powertools-python/pull/971/files#diff-be72cf49c5b9ee9da9930ced7b33c6112dcc741221f3dfa0579b99e15fc8e8ffR145
> 
> I'm not entirely sure, perhaps what the `THEN` line should look like?

`test_copy_config_to_ext_loggers_clean_old_handlers` test checks if external logger was reconfigured to use new handler and formatter and that old configured handler `handler = logging.FileHandler("logfile")` was cleaned

**Comment by michaelbrewer** [comment]
@mploski @heitorlessa - is this PR cool as it is? Or are there other changes needed?

**Comment by mploski** [comment]
> @mploski @heitorlessa - is this PR cool as it is? Or are there other changes needed?

Looks good to me :)

**Comment by michaelbrewer** [comment]
@heitorlessa - anything outstanding?

**Comment by heitorlessa** [comment]
> @heitorlessa - anything outstanding?

Michal comes back from PTO tomorrow and will make suggestions to fix his test comments to make the intent clearer - then we can merge (instead of creating another PR after merging this)

**Comment by mploski** [comment]
@michaelbrewer I reviewed test's comments I added some time ago for these tests and definitely I could have made them better :-) If you could tune them that would be great! Here are some suggestions, feel free to modify it :-) 

**test_copy_config_to_ext_loggers**
Line 53 -> # GIVEN  two external loggers and powertools logger initialized
Line 59 -> # WHEN configuration copied from powertools logger to ALL external loggers AND external loggers used
Line 66 -> # THEN all external loggers used Powertools handler, formatter and log level

**test_copy_config_to_ext_loggers_include**
Line 81 -> # WHEN configuration copied from powertools logger to INCLUDED external loggers AND our external logger used 
Line 87 -> # THEN included external loggers used Powertools handler, formatter and log level. 

**test_copy_config_to_ext_loggers_wrong_include**
 Line 101 -> # WHEN configuration copied from powertools logger to INCLUDED NON EXISTING external loggers
Line 104 -> # THEN existing external logger is not modified

**test_copy_config_to_ext_loggers_exclude**
Line 113 -> # WHEN configuration copied from powertools logger to ALL BUT external logger
Line 116 -> #  THEN external logger is not modified

**test_copy_config_to_ext_loggers_include_exclude**
Line 121 -> # GIVEN two external loggers and powertools logger initialized
Line 127 -> # WHEN configuration copied from powertools logger to INCLUDED external loggers AND external logger_1 is also in EXCLUDE list
Line 135 -> # THEN logger_1 is not modified and Logger_2 used Powertools handler, formatter and log level

**test_copy_config_to_ext_loggers_clean_old_handlers**
Line 152 ->  # WHEN configuration copied from powertools logger to ALL external loggers
Line 155 -> # THEN old logger handler is removed and Powertools configuration is used instead

**test_copy_config_to_ext_loggers_custom_log_level**
Line 167 -> # WHEN configuration copied from powertools logger to INCLUDED external logger AND external logger used with custom log_level
Line 174 -> # THEN  external logger used Powertools handler, formatter and CUSTOM log level.

File: tests/functional/test_logger_utils.py [hunk 1]
```diff
@@ -51,142 +50,134 @@ def service_name():
...
 
 def test_copy_config_to_ext_loggers_wrong_include(stdout, logger, log_level):
-
-    # GIVEN a external logger and powertools logger initialized
+    # GIVEN an external logger and powertools logger initialized
     logger = logger()
     powertools_logger = Logger(service=service_name(), level=log_level.INFO.value, stream=stdout)
 
-    # WHEN configuration copied from powertools logger to ALL external loggers AND our external logger used
+    # THEN included external loggers used Powertools handler, formatter and log level.
```
Comment (mploski) [inline tests/functional/test_logger_utils.py:103]: Shouldn't it be something like `WHEN configuration copied from powertools logger to INCLUDED NON EXISTING external loggers`?

**Review by mploski — APPROVED** [review]
Looks great!  Let's merge it

**Pull request merged**